### DISCUSSION TITLE
User can try modifying "Last update" date of record

### DISCUSSION THREAD

**Discussion by BOHEUS** [twentyhq/twenty#7021] [Ideas]
**Scenario:**
1. Log in
2. Go to any object's record details (e.g. Companies)
3. Click on "Last update" date

**Actual:** After clicking on "Last update" date, calendar appears and user can try modifying the date
[Screencast from 2024-08-31 21-34-41.webm](https://github.com/user-attachments/assets/9a0eb7de-c9e9-4a21-9771-03a77ed53f98)

**Expected:** User shouldn't be able to modify the "Last update" date as it's automatically modified by the system

**Comment by charlesBochet** [comment]
@BOHEUS Yes, we are aware of it!

@FelixMalfait @Bonapara, it would be great to take a product decision about the "isSystem" boolean:

Naive proposal:
Objects:
- isSystem still LGTM to be honnest

Fields:
- isSystem => isReadonly? and we leave users displaying any field they want except for relations with system objects

**Comment by FelixMalfait** [comment]
@charlesBochet There are several cases where we've been asked to create a field that could be changed in the UI (e.g. a field that the workspace admin decides to update via API only).

I agree we need to create a new attribute at the field level. I think we should keep "ReadOnly" for the permission-world, it will be possible to enforce stricter rules once we have them. So we could call this one `isEditable`?  It should also be displayed to user in settings, I think @Bonapara already did mocks for that

That will not solve getting rid of `isSystem`, but at least we're not adding more debt to it :)

**Comment by Bonapara** [comment]
Related to https://github.com/twentyhq/twenty/issues/6491

**Comment by FelixMalfait** [comment]
OK discussed with Thomas and we will solve this at the View level + Layout level for records. 
This is either a permission issue or a layout issue, but we think it's best not to do it at the data-model level.

So it means:
- Introducing `isEditable` column in `ViewField` ("Views" are the layouts for lists)
- Introducing isEditable when we have the layout system for record pages

**Comment by FelixMalfait** [comment]
Converting to a discussion as this is a long-term issue

**Comment by BOHEUS** [comment]
Now it's not possible anymore to update "Last update" field
